htmresearch/encoders/language_encoder.py

```diff
@@ -183,11 +183,14 @@ def sparseUnion(self, counts):
 
     @param counts     (Counter)   A count of the ON bits for the union bitmap.
 
-    @return           (list)      A sparsified union bitmap.
+    @return           (tuple)      A sparsified union bitmap.
     """
+    positionsSortedByCount = tuple(x[1] for x in counts.getNonZerosSorted())
```

Comment:
I would actually prefer a random tie breaker.  Otherwise we are biasing the system to prefer bits near the top of the retina.  One way is to add a small random number to every element before sorting, similar to what we did with the spatial pooler.